src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2452,6 +2451,35 @@ bool learnFromOperator(BoundBinaryOperator binary)
 
                 return false;
             }
+
+            // This function is only for the specific scenario where the LHS of a binary '=='/'!=' operator is a conditional access
+            // and the RHS is either a constant value or is of a non-nullable value type.
```

Comment:
Consider asserting these conditions.

Reply:
I found it prohibitively cumbersome to assert on this meaningfully, so I ended up deciding to just inline the local function at its call site.